### PR TITLE
Notes view autosave no longer removes changes on notes being edited

### PR DESCRIPTION
**Pull request by altair1673**
### Prerequisites

- [x] Reviewed the [checklist](CHECKLIST.md)

- [ ] Reviewed feedback from the "Sonar Cloud" bot. Note that you have to wait
    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change

**_Fix for issue #1848 found during testing of release candidate v2.10.0-rc1_**

Notes that are being edit do not get reset when an autosave occurs. If a note is in edit mode and the user switches graphs, that note will remain in edit mode with any changes made when the user returns.

### Why Should This Be In Core?

Makes editing notes safe as the users can the sure there changes won't randomly be lost if a background plugin finishes running.
### Benefits

Increases usability of the notes view.
### Possible Drawbacks

Auto notes are not updated in real time.
### Verification Process

Made changes to note in edit mode switched graphs and came back to see note still in edit mode with the changes I have made.
### Applicable Issues

#1848

### CODE REVIEW

File: CoreNotesView/src/au/gov/asd/tac/constellation/views/notes/state/NotesViewEntry.java
Comment (OrionsGuardian) [inline CoreNotesView/src/au/gov/asd/tac/constellation/views/notes/state/NotesViewEntry.java:160]: final

File: CoreNotesView/src/au/gov/asd/tac/constellation/views/notes/state/NotesViewEntry.java
Comment (OrionsGuardian) [inline CoreNotesView/src/au/gov/asd/tac/constellation/views/notes/state/NotesViewEntry.java:168]: final

File: CoreNotesView/src/au/gov/asd/tac/constellation/views/notes/state/NotesViewEntry.java
Comment (OrionsGuardian) [inline CoreNotesView/src/au/gov/asd/tac/constellation/views/notes/state/NotesViewEntry.java:176]: final

**Review by OrionsGuardian — CHANGES_REQUESTED** [review]

**Review by Delphinus8821 — APPROVED** [review]

**Review by OrionsGuardian — APPROVED** [review]

File: CoreNotesView/src/au/gov/asd/tac/constellation/views/notes/NotesViewPane.java
Comment (antares1470) [inline CoreNotesView/src/au/gov/asd/tac/constellation/views/notes/NotesViewPane.java:587]: use a for each here `for (final NotesViewEntry note : notesToRender)`

File: CoreNotesView/src/au/gov/asd/tac/constellation/views/notes/state/NotesViewEntry.java
Comment (antares1470) [inline CoreNotesView/src/au/gov/asd/tac/constellation/views/notes/state/NotesViewEntry.java:47]: add the instantiations to the constructor

**Review by antares1470 — CHANGES_REQUESTED** [review]

**Review by antares1470 — APPROVED** [review]
Good to go once the merge conflict is resolved

**Pull request merged**
